Makefile

```diff
@@ -0,0 +1,59 @@
+VERSION=$(shell git describe --tags --always)
+COMMIT=$(shell git rev-parse HEAD)
+BUILD=$(shell date +%FT%T%z)
+PKG=github.com/guacsec/guac
+
+LDFLAGS="-X $(PKG).version=$(VERSION) -X $(PKG).commit=$(COMMIT) -X $(PKG).date=$(BUILD)"
+
+.PHONY: all
+all: setup test cover fmt lint ci build
+    
+.PHONY: setup
+setup: ## Install all the build and lint dependencies
+	curl -L https://git.io/vp6lP | sh
+
+
+.PHONY: test
+test: ## Run all the tests
+	echo 'mode: atomic' > coverage.txt && go test -covermode=atomic -coverprofile=coverage.txt -v -race -timeout=30s ./...
+
+
+.PHONY: cover
+cover: test ## Run all the tests and opens the coverage report
+	go tool cover -html=coverage.txt
+
+.PHONY: fmt
+fmt: ## Run goimports on all go files
+	find . -name '*.go' -not -wholename './vendor/*' | while read -r file; do goimports -w "$$file"; done
+
+.PHONY: lint
+lint: ## Run all the linters
```

Comment:
i think i'd personally prefer golangci-lint here. seems like a more popular choice recently.

Reply:
golangci-lint instead of `gometalinter` ? I've no particular preference, I've used metalinter, but open to add ci-lint.

Reply:
yep!